File: latihan pemrograman dasar/calc.py
import math

history = []

print("=== Kalkulator Python ===")

while True:
    print("\nOperasi: + - * / ** sqrt | q untuk keluar")
    op = input("Pilih: ")

    if op == "q":
        break

    try:
        if op == "sqrt":
            x = float(input("Angka: "))
            result = math.sqrt(x)
            history.append(f"sqrt({x}) = {result}")
            print("Hasil:", result)

        else:
            a = float(input("A: "))
            b = float(input("B: "))

            if op == "+":
                result = a + b
            elif op == "-":
                result = a - b
            elif op == "*":
                result = a * b
            elif op == "/":
                result = a / b if b != 0 else "Error (0)"
            elif op == "**":
                result = a ** b
            else:
                print("Operator salah")
                continue

            history.append(f"{a} {op} {b} = {result}")
            print("Hasil:", result)

    except ValueError:
        print("Input harus angka!")

print("\n=== History ===")
for h in history:
    print(h)

File: latihan pemrograman dasar/calc.c
#include <stdio.h>

int main() {
    char op;
    float a, b;
    int running = 1;

    printf("=== Kalkulator C ===\n");

    while (running) {
        printf("\nMasukkan operasi (+, -, *, /) atau q untuk keluar:\n");

        scanf(" %c", &op);

        if (op == 'q') {
            running = 0;
            break;
        }

        printf("Masukkan angka (format: a b): ");
        scanf("%f %f", &a, &b);

        switch (op) {
            case '+':
                printf("Hasil: %.2f\n", a + b);
                break;
            case '-':
                printf("Hasil: %.2f\n", a - b);
                break;
            case '*':
                printf("Hasil: %.2f\n", a * b);
                break;
            case '/':
                if (b != 0)
                    printf("Hasil: %.2f\n", a / b);
                else
                    printf("Error: pembagian 0\n");
                break;
            default:
                printf("Operator tidak valid\n");
        }
    }

    printf("Program selesai\n");
    return 0;
}